kf/processCheck.go

```diff
@@ -59,8 +59,25 @@ func (c *pCheck) pMonitorWorker(bpfProgs map[string]*list.List, direction string
 					bpf.RestartCount++
 					log.Warn().Msgf("pMonitor BPF Program is not running. Restart attempt: %d, program name: %s, iface: %s",
 						bpf.RestartCount, bpf.Program.Name, ifaceName)
-					if err := bpf.Start(ifaceName, direction, c.Chain); err != nil {
-						log.Error().Err(err).Msgf("pMonitor BPF Program start failed for program %s", bpf.Program.Name)
+					//  User program is a daemon and not running, but the BPF program is loaded
+					if !userProgram && bpfProgram {
+						if err := bpf.StartUserProgram(ifaceName, direction, c.Chain); err != nil {
```

Comment:
Here we restart the user program, but the bpfProgram was loaded.  Is it safe to restart the user program without unload (may not be possible) bpfProgram?

Reply:
This scenario is when user program is a daemon and stopped. Then l3afd will restart only when it is down. It is not necessary to reload bpfprogram.